xdem/spatialstats.py

```diff
@@ -735,47 +819,171 @@ def sample_empirical_variogram(values: Union[np.ndarray, RasterType], gsd: float
+            elif model_name in ['stable', 'matern']:
+                s = params_vgm['smooth'].values[i]
+                fn += model_function(h, r, p, s)
+        return fn
+
+    return sum_model
+
+def covariance_from_vgm(params_vgm: pd.DataFrame) -> Callable[[np.ndarray], np.ndarray]:
+    """
+    Construct the spatial covariance function from a dataframe of variogram parameters.
+    The covariance function is the sum of partial sills "PS" minus the sum of associated variograms "gamma":
+    C = PS - gamma
+
+    :param params_vgm: Dataframe of variogram models to sum with three to four columns, "model" for the model types
+        (e.g., ["spherical", "matern"]), "range" for the correlation ranges (e.g., [2, 100]), "psill" for the partial
+        sills (e.g., [0.8, 0.2]) and "smooth" for the smoothness paramter if exists for this model (e.g., [None, 0.2]).
+
+    :return: Covariance function with spatial lags
+    """
+
+    # Check input dataframe
+    _check_validity_params_vgm(params_vgm)
+
+    # Get total sill
+    total_sill = np.sum(params_vgm['psill'])
+
+    # Get function from sum of variogram
+    sum_vgm = get_func_sum_vgm_models(params_vgm)
+
+    def cov(h):
+        return total_sill - sum_vgm(h)
+
+    return cov
+
+def correlation_from_vgm(params_vgm: pd.DataFrame)-> Callable[[np.ndarray], np.ndarray]:
     """
-    Fit a multi-range variogram model to an empirical variogram, weighted least-squares based on sampling errors
+    Construct the spatial correlation function from a dataframe of variogram parameters.
+    The correlation function is the covariance function "C" divided by the sum of partial sills "PS": rho = C / PS
+
+    :param params_vgm: Dataframe of variogram models to sum with three to four columns, "model" for the model types
+        (e.g., ["spherical", "matern"]), "range" for the correlation ranges (e.g., [2, 100]), "psill" for the partial
+        sills (e.g., [0.8, 0.2]) and "smooth" for the smoothness paramter if exists for this model (e.g., [None, 0.2]).
+
+    :return: Correlation function with spatial lags
+    """
+
+    # Check input dataframe
+    _check_validity_params_vgm(params_vgm)
 
-    :param list_model: list of K variogram models to sum for the fit: from short-range to long-ranges
-    :param empirical_variogram: empirical variogram
-    :param bounds: bounds of ranges and sills for each model (shape K x 4 = K x range lower, range upper, sill lower, sill upper)
-    :param p0: initial guess of ranges and sills each model (shape K x 2 = K x range first guess, sill first guess)
+    # Get total sill
+    total_sill = np.sum(params_vgm['psill'].values)
 
-    :return: modelled variogram function, coefficients
+    # Get covariance from sum of variogram
+    cov = covariance_from_vgm(params_vgm)
+
+    def rho(h):
+        return cov(h)/total_sill
+
+    return rho
+
+
+def fit_sum_model_variogram(list_models: list[str] | list[Callable], empirical_variogram: pd.DataFrame,
```

Comment:
Some functions use the term `vgm`, some `variogram`. We should try to be consistent. 
Maybe rename `fit_sum_vgm_models`?

Reply:
Renamed everything with `variogram`, 6 letters doesn't make much of a difference so I went for clarity! :smile: